### PR TITLE
Remove unnecessary transmute calls

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
I removed some `transmute` calls and added comments where they were missing.

cc @sdroege @EPashkin

### CODE REVIEW

File: src/object.rs [hunk 1]
```diff
@@ -1541,6 +1541,7 @@ impl<T: ObjectType> ObjectExt for T {
         ::signal::connect_raw(
             self.as_object_ref().to_glib_none().0,
             signal_name.as_ptr() as *const _,
+            // Two level-kind of callback, which explains why the transmute is required.
```
Comment (sdroege) [inline src/object.rs:1544]: I don't understand the comment

Comment (GuillaumeGomez) [reply src/object.rs:1544]: I might have been a bit too lazy writing this comment, let me improve it.

Comment (sdroege) [reply src/object.rs:1544]: Same for the other such comments. What's "two levels" here exactly, or "two times" (below)?

File: src/object.rs [hunk 2]
```diff
@@ -1541,7 +1541,7 @@ impl<T: ObjectType> ObjectExt for T {
         ::signal::connect_raw(
             self.as_object_ref().to_glib_none().0,
             signal_name.as_ptr() as *const _,
-            Some(mem::transmute(notify_trampoline::<Self, F> as usize)),
+            Some(*(&notify_trampoline::<Self, F> as *const _ as *const _)),
```
Comment (EPashkin) [inline src/object.rs:1544]: Interesting how rust found type for first `_` in `as *const _ as *const _ `  😉

**Comment by EPashkin** [comment]
👍

File: src/object.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/object.rs:1544]: Pointers are a weird thing in rust. You have to do it twice every time to convert between pointers of non primitives types.

Comment (sdroege) [reply src/object.rs:1544]: The first `as *const _` casts the `&T` into a `*const T` (there is only one option for `T` here, you can't do transitive casts from references). The second `as *const _` is from `*const T` to `*const glib_sys::GCallback` (which is what is required to be the final type).

Comment (EPashkin) [reply src/object.rs:1544]: Thanks

**Pull request merged**